src/pyglottolog/api.py

```diff
@@ -25,6 +26,30 @@
 ISO_CODE_PATTERN = re.compile('[a-z]{3}$')
 
 
+class Cache:
+    def __init__(self):
+        self.by_iso = {}
+        self.by_id = {}
+        self._lineage = {}
+
+    def __contains__(self, item):
+        return (item in self.by_iso) or item in self.by_iso
+
```

Comment:
I we stick to two `dicts`, this (and the `__getitem__` below) might be a good use case for a [collections.ChainMap](https://docs.python.org/3/library/collections.html#collections.ChainMap)

Call the argument here and below `id_` (see further below, or `key`)?

Reply:
Ah, never use a `ChainMap`. The only disadvantage I'd see is that it makes the function of the two dicts less transparent. And adding stuff like
```python
@property
def by_iso(self):
    return self.maps[1]
```
eats up a lot of the saved lines of code :)

Reply:
Hm, the way I check for the existence of a cache will require another magic method on `Cache`
```python
def __bool__(self):
    return True
```
further diminishing the gain of switching to `ChainMap`. So, I'm not really convinced - although it feels tempting to use something new, which is "almost exactly what I need" :)

Reply:
Not sure, I understand why using a `ChainMap` is less transparent from the example above: I was assuming the `dict`s here can be private, and we could have both a reference to the `ChainMap` as well as to the indivudual dicts (still not sure whether we need two dicts as their key-spaces are disjoint, see my question above)

Reply:
The `dict`s for ISO codes and Glottocodes don't have a clear hierarchy, regarding which one to lookup first - as for most other examples of `ChainMap` I've seen so far. So to me, using `ChainMap` or not comes down to whether I can save code or not. Since I'd have to overwrite `__bool__` to save `__getitem__` with `ChainMap`, I opted for not changing.

Reply:
Okay. For me this would be less about saving keystrokes and more about correctness and explicitness. My understanding was that there is a clear hierarchy (by_iso > by_id) and this was repeated which coupld be abstracted with a `ChainMap?`

Now I am not sure any more if this is correct (should the second be `by_id`)?

```python
         return (id_ in self.by_iso) or id_ in self.by_iso
```